edgelet/iotedge/src/check/checks/up_to_date_config.rs

```diff
@@ -0,0 +1,96 @@
+use std::fs;
+use std::path::Path;
+
+use failure::{self, Context, Fail};
+
+use crate::check::{checker::Checker, Check, CheckResult};
+
+#[derive(Default, serde_derive::Serialize)]
+pub(crate) struct UpToDateConfig {}
+
+impl Checker for UpToDateConfig {
+    fn id(&self) -> &'static str {
+        "aziot-configs-up-to-date"
+    }
+    fn description(&self) -> &'static str {
+        "aziot configurations up-to-date with config.toml"
+    }
+    fn execute(&mut self, check: &mut Check, _: &mut tokio::runtime::Runtime) -> CheckResult {
+        Self::inner_execute(check).unwrap_or_else(CheckResult::Failed)
+    }
+    fn get_json(&self) -> serde_json::Value {
+        serde_json::to_value(self).unwrap()
+    }
+}
+
+impl UpToDateConfig {
+    fn inner_execute(_check: &mut Check) -> Result<CheckResult, failure::Error> {
+        let config = Path::new("/etc/aziot/config.toml");
+
+        if !config.exists() {
+            return Ok(CheckResult::Ignored);
+        }
+
+        let config_metadata = match fs::metadata(config) {
+            Ok(m) => m,
+            Err(err) => {
+                return Ok(CheckResult::Failed(
+                    err.context(format!("Failed to query metadata of {}", config.display()))
+                        .into(),
+                ))
+            }
+        };
+
+        let config_last_modified = config_metadata
+            .modified()
+            .expect("file metadata should contain valid last_modified");
+
+        for service in &["keyd", "certd", "identityd", "tpmd", "edged"] {
```

Comment:
Implement the keyd/certd/identityd/tpmd checks in i-i-s, not here.

Reply:
So move this entire check to i-i-s? Since the keyd/certd/identityd/tpmd checks are the majority of this check.

Reply:
`aziotctl check` will have a check that checks the files for the i-i-s services. `iotedge check` will have a check that checks the file for edged. You can move the loop body to a function in `aziotctl-common` so that both checks can share it.

Reply:
Got it. Is there an easy way for a function in `aziotctl-common` to return the `CheckResult` type? I see that there's a different `CheckResult` for i-i-s and iotedge.

Reply:
You can return the new `CheckResult` and manually map the variants to the old `CheckResult` - `anyhow::Error` -> `failure::Error` conversion is `Context::new(format!("{:?}", err)).into()` (at the cost of flattening the context chain into a single error). It's hacky but it'll all get cleaned up when we drop `failure` anyway.

Or you can just duplicate the code instead of making a common function.